test/js/service/PiskelApiServiceTest.js

```diff
@@ -0,0 +1,281 @@
+      expect(piskelWindow.parent.postMessage.calls.argsFor(0)[0].type).toBe(PiskelApi.MessageType.PISKEL_API_READY);
+    });
+
+    it("adds an event listener on its own window", function() {
+      expect(piskelWindow.addEventListener.calls.count()).toEqual(1);
+    });
+
+    it("subscribes to events", function() {
+      expect($.subscribe.calls.count()).toEqual(5);
+      expect($.subscribe.calls.argsFor(0)[0]).toBe(Events.PISKEL_SAVE_STATE);
+      expect($.subscribe.calls.argsFor(1)[0]).toBe(Events.FPS_CHANGED);
+      expect($.subscribe.calls.argsFor(2)[0]).toBe(Events.HISTORY_STATE_LOADED);
+      expect($.subscribe.calls.argsFor(3)[0]).toBe(Events.FRAME_SIZE_CHANGED);
+      expect($.subscribe.calls.argsFor(4)[0]).toBe(Events.ADD_NEW_FRAME_CLICKED);
+    });
+  });
+
+  describe('createNewPiskel()', function() {
+    var service, piskelController;
+    beforeEach(function() {
+      piskelController = createFakePiskelController();
+      service = new pskl.service.PiskelApiService(piskelController);
+      service.init(piskelWindow);
+    });
+
+    afterEach(function() {
+      piskelController.setPiskel.calls.reset();
+    });
+
+    it("sends an ANIMATION_LOADED message once creating a new piskel", function() {
+      service.createNewPiskel(100, 100, 4);
+      // One message for init and one message for create new.
+      expect(piskelWindow.parent.postMessage.calls.count()).toEqual(2);
+      expect(piskelWindow.parent.postMessage.calls.argsFor(1)[0].type).toBe(PiskelApi.MessageType.ANIMATION_LOADED);
+    });
+
+    it("sets a new piskel with a single frame and layer", function() {
+      service.createNewPiskel(100, 100, 4);
+      expect(piskelController.setPiskel.calls.count()).toBe(1);
+      expect(piskelController.setPiskel.calls.argsFor(0)[0].width).toBe(100);
+      expect(piskelController.setPiskel.calls.argsFor(0)[0].height).toBe(100);
+      expect(piskelController.setPiskel.calls.argsFor(0)[0].fps).toBe(4);
+      expect(piskelController.setPiskel.calls.argsFor(0)[0].descriptor.name).toBe('New Piskel');
+      expect(piskelController.setPiskel.calls.argsFor(0)[0].layers.length).toBe(1);
+      expect(piskelController.setPiskel.calls.argsFor(0)[0].layers[0].frames.length).toBe(1);
+    })
+  });
+
+  describe('loadSpritesheet()', function() {
+    var service, piskelController, importService, imageObject, onePixelBlackURI, imageSpy;
+
+    beforeEach(function() {
+      piskelController = createFakePiskelController();
+      importService = createFakeImportService();
+      service = new pskl.service.PiskelApiService(piskelController, {}, importService);
+      service.init(piskelWindow);
+
+      imageSpy = spyOn(window, "Image").and.callFake(function() {
+        imageObject = this;
+        this.onload = function() {};
+        spyOn(imageObject, 'onload');
+      });
+
+      onePixelBlackURI = "image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAAAAAA6fptVAAAACklEQVR4nGNiAAAABgADNjd8qAAAAABJRU5ErkJggg==";
```

Comment:
As a constant you can (probably should) define this outside of the `beforeEach` function.